v2/http-middleware/security.go

```diff
@@ -150,19 +158,11 @@ func getUserIDByToken(ctx context.Context, accessToken string) (_ string, err er
 		return
 	}
 
-	req, err := http.NewRequest(http.MethodGet, baseURL+endpoint, nil)
-	if err != nil {
-		err = errors.Wrap(err, "failed to create new HTTP request")
-		return
-	}
-
-	req = req.WithContext(ctx)
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Authorization", accessToken)
-
-	client := &http.Client{Transport: &ochttp.Transport{}}
+	req := rest.Get(baseURL+endpoint).
+		SetHeader("Content-Type", "application/json").
+		SetHeader("Authorization", accessToken)
 
-	resp, err := client.Do(req)
+	resp, err := client.Do(ctx, req)
 	if err != nil {
```

Comment:
The error handling below will not be the same with the old and the new rest client.

The new rest client treats an non-200 response as an error.